#1 안드로이드 초기 세팅

ricky-o-k opened the pull request:
#1

ricky-o-k committed:
[base] : 데이터바인딩을 사용하는 Base 화면 및 템플릿 코드 작성 (#1)

사용 예는 example 패키지 내용 참고

ricky-o-k committed:
[base] : ViewModel 과 데이터바인딩을 사용하는 Base 화면 및 템플릿 코드 작성 (#1)

사용 예는 example/___/viewmodel 패키지 내용 참고

ricky-o-k committed:
[base] : github 연동 테스트 (#1)

ricky-o-k committed:
[base] : GitHub actions CI 시스템 구축 (#1)

자세한 내용은 .github/workflows/android.yml 파일 내용 참고

ricky-o-k committed:
[base] : 초기 테스트 환경 구축 & viewModel 접근제한자 수정 (#1)

androidx.test, espresso, junit, mockK, robolectric

ricky-o-k committed:
[bug] : ci 오류 내용 수정 (#1)

ricky-o-k committed:
[bug] : ci 오류 내용 수정 - java 버전 9로 수정 (#1)

ricky-o-k committed:
[bug] : ci 오류 내용 수정 - java-package 명시 (jdk) (#1)

ricky-o-k committed:
[bug] : ci 오류 내용 수정 - sdk version 28로 강제 조정 (#1)

github actions 에서 아직 Java 9 지원이 미흡한듯하여 로컬에서만 Java 9로 테스트 진행

ricky-o-k commented:
일단 기본적인 화면 작업 기능만 올려둔 상태여서 (MVVM, DataBinding) 
아직 코루틴이라거나 Rx 에서 필요한 내용들은 넣지 않은 상태입니다. 일부 내용들은 리뷰하면서 넣어볼까 생각했어서..! 

현재 이걸로 MVVM, DataBinding 확인해보시고 추가하고 싶은 내용이나 잘못 작성되어 있거나 하는 내용 말씀해주시면 될 것 같고
위에서 언급했던 내용 (Rx 등) 에 대해서도 이야기 되면 좋을 것 같아요 ~ _ ~

riflockle7 commented:
slack PR 메시지 테스트

yunakim2 reviewed (commented):
주석까지 모두 완벽해서 사실 코드에 관한 리뷰를 해드릴 게 없네요,, 너무 고생하셨어요 👍🏻

ricky-o-k commented:
코드리뷰 관련 3가지 방법 : https://www.notion.so/406b5cac78874e8da23a034ac0d7992a

riflockle7 commented on app/src/main/java/kr/co/nexters/winepick/WinePickApplication.kt:
```diff
@@ -0,0 +1,15 @@
+/**
+ * WinePick Application 클래스
+ *
+ * @author ricky
```
이 파일에 이름이 빠지거나 아니면 새로 규칙을 만들거나 해야할 것 같아요

riflockle7dummy replied:
@author 을 빼거나 각 파일에 추가하거나 선택해야 될 것 같습니다.

riflockle7dummy reviewed (changes requested):
Request changes 의 예

ricky-o-k commented on app/src/main/java/kr/co/nexters/winepick/example/java/normal/JavaDefaultFragment.java:
```diff
@@ -0,0 +1,39 @@
+        super.onViewCreated(view, savedInstanceState);
+
+        // 텍스트 값 세팅
+        binding.tvTitle.setText("자바 프레그먼트 테스트");
```
문자열을 `하드코딩`이 아닌 다른 방식으로 하면 어떨까요?

ricky-o-k reviewed (commented):
comment 의 예

ricky-o-k committed:
[base] : 1. kotlinx.serialization 적용

GSON 과 비슷한 직렬화/역직렬화 라이브러리

ricky-o-k committed:
[base] : 2. Ktor 적용

Retrofit 과 비슷한 네트워크 통신 라이브러리

GET 과 POST 까지 테스트 완료

ricky-o-k committed:
[base] : 3. SharedPreference

SharedPreference 를 map 처럼 사용할 수 있도록 함

ricky-o-k committed:
[base] : 4. Retrofit2

사용 예제는 Retrofit2Test.kt 내용 참고

같이 import 한 라이브러리
- okhttp3:logging-interceptor
- retrofit2-kotlinx-serialization-converter
- retrofit2:adapter-rxjava2

ricky-o-k committed:
[base] : 논의한대로 사용 / 미사용 라이브러리 및 테스트 코드 위치 옮김

deprecated 패키지에 두어 사용 예제는 남겨놓음

ricky-o-k committed:
Merge pull request #4 from Nexters/base/init-riflockle7

2020.02.04 회의에서 나온대로 기술 반영 & Retrofit 기초모듈 구축

ricky-o-k committed:
[base] : RxJava3 를 이용하는 경우, Coroutine 을 이용하는 경우의 코드 작성

Base 에 RxJava3, Coroutine 과 관련된 내용 추가함
